src/arithmetic/arithmetic_expression.c

```diff
@@ -558,11 +558,12 @@ void AR_EXP_ToString(const AR_ExpNode *root, char **str) {
 
 // Generate a heap-allocated name for an arithmetic expression.
 // This routine is only used to name ORDER BY expressions.
-void AR_EXP_BuildResolvedName(AR_ExpNode *root) {
-	if(root->resolved_name) return;
+char *AR_EXP_BuildResolvedName(AR_ExpNode *root) {
+	if(root->resolved_name) return NULL;
 	char *new_name = (char *)root->resolved_name;
 	AR_EXP_ToString(root, &new_name); // Note that this string must be freed!
 	root->resolved_name = new_name;
+	return new_name;
```

Comment:
`root->resolved_name` is `const`, and we can't cast _and_ take the address inline here. I'd like to keep that member const, since this is the only scenario in which it isn't.

Reply:
Disgregard this comment, I was overlooking something simple! :+1: